xbmc/addons/settings/AddonSettings.cpp

```diff
@@ -1319,17 +1319,18 @@ bool CAddonSettings::ParseOldLabel(const TiXmlElement* element, const std::strin
   labelId = -1;
   if (element == nullptr)
     return false;
+  
+  // label value as a string
+  std::string labelString;
+  element->QueryStringAttribute("label", &labelString);
+
 
   // try to parse the label as a translation number
-  if (element->QueryIntAttribute("label", &labelId) == TIXML_SUCCESS && labelId >= 0)
+  if (element->QueryIntAttribute("label", &labelId) == TIXML_SUCCESS &&
```

Comment:
I would use `strtol()` to try to parse the `labelString` as an integer. If the `endptr` parameter is either `nullptr` or `'\0'` the `"label"` attribute is a pure number.